fix transform for rcnns so original images list is unchanged

buoyancy99 opened the pull request:
Maskrcnn or fastrcnn pass input image list into the transform module in torchvision/models/detection/transform.py. However, the method modifies content in the list where transformed images are pasted back to corresponding entries of the input list.  

As a result, user may meet unintended behavior when they hold reference to the original input list outside the maskrcnn call. The proposed change leave user input image list to the maskrcnn call unchanged.

fmassa commented on torchvision/models/detection/transform.py:
```diff
@@ -30,17 +30,19 @@ def __init__(self, min_size, max_size, image_mean, image_std):
+        transformed_images = []
         for i in range(len(images)):
-            image = images[i]
+            image = images[i].clone()
```
This clone is not necessary because `normalize` is done out-of-place

buoyancy99 replied:
removed

fmassa commented on torchvision/models/detection/transform.py:
```diff
@@ -30,17 +30,19 @@ def __init__(self, min_size, max_size, image_mean, image_std):
         self.image_std = image_std
 
     def forward(self, images, targets=None):
+        transformed_images = []
```
A simpler way of doing this (and a one-liner) would be
```python
images = [img for im in images]
```

buoyancy99 replied:
Done with list slicing.

buoyancy99 commented:
> Thanks for the PR!
> 
> I have a few comments.
> 
> Also, could you add tests for this case? Maybe in `tests/test_models.py`

fixed and added test in #1085
